core/src/textcodec/GBTextEncoder.cpp

```diff
@@ -4000,13 +4000,13 @@ void GBTextEncoder::EncodeGB18030(const std::wstring& str, std::string& bytes)
 					bytes[index++] = replacement;
 					++invalid;
 				}
-				high = -1;
+				high = static_cast<unsigned>(-1);
```

Comment:
I believe that was a bug that `high` is `unsigned`.

Reply:
I do not have enough knowledge about the overall design.
But this is a point in favor of resolving warnings :)

Reply:
Yep, definitively a bug as seen from [Qt code](https://github.com/qt/qtbase/blob/dev/src/corelib/codecs/qgb18030codec.cpp#L88-L144). The fix it either leave `high` declared as `unsigned` and change here and below to `high = 0`, or change the declaration of high `int high = -1`. In either case, no cast is needed.

Reply:
Ok @huycn, feel free to resolve this conversation.

Reply:
I've pushed a fix. You can resolve if you find it's ok.